packages/block-library/src/paragraph/edit.js

```diff
@@ -147,7 +148,7 @@ function ParagraphBlock( {
 					<RichText
 						ref={ ref }
 						identifier="content"
-						tagName="p"
+						tagName={ Block.p }
```

Comment:
It's an enhanced tag name. It's still explicitly sets it to a paragraph, but the paragraph is also a block. We've done it before with the animated component from `react-spring`:

https://github.com/WordPress/gutenberg/blob/60ff7a78550a4eb93179bbcf1ec06478df9495e0/packages/block-editor/src/components/block-list/block.js#L474-L484

> Would it make sense to have a flag like isBlockWrapper in RichText and simply pass the tagName as we do now, but when the flag is enabled RichText would then use BlockComponent/BlockWrapper with the tag-name provided automatically?

I don't find this solution as nice. `RichText` should ideally not know anything about blocks. The `tagName` prop is meant to render the component provided, whether it's simply an element component (shortcut by providing just a string) or a custom component. It works without any changes to `RichText`.

Perhaps the prop name could be improved, but I think <code><strong>tagName</strong>: Block.<strong>p</strong></code> makes sense.

Reply:
Thank you for clarifying this @ellatrix. I think we can keep things as they are 👍 In the future it would be good to rename tagName to indicate any component is ok.